src/nwp_consumer/internal/inputs/ecmwf/client.py

```diff
@@ -0,0 +1,243 @@
+                return []
+
+            if os.stat(tf.name).st_size < 100 and "0 bytes" in tf.read():
+                return []
+
+        return [ECMWFMarsFileInfo(inittime=it, area=self.area)]
+
+    def downloadToTemp(self, *, fi: internal.FileInfoModel) \
+            -> tuple[internal.FileInfoModel, pathlib.Path]:
+        tfp: pathlib.Path = internal.TMP_DIR / fi.filename()
+        try:
+            self.server.execute(
+                req=f"""
+                    retrieve,
+                        class    = od,
+                        date     = {fi.it().strftime("%Y%m%d")},
+                        expver   = 1,
+                        levtype  = sfc,
+                        param    = {'/'.join(list(PARAMETER_ECMWFCODE_MAP.keys()))},
+                        step     = 0/to/48/by/1,
+                        stream   = oper,
+                        time     = {fi.it().strftime("%H")},
+                        type     = fc,
+                        area     = {AREA_MAP[self.area]},
+                        grid     = 0.05/0.05,
+                        target   = off
+                    """,
+                target=tfp.as_posix()
+            )
+        except ecmwfapi.api.APIException as e:
+            log.warn("error fetching ECMWF MARS data", error=e)
+            return fi, pathlib.Path()
+
+        log.debug(
+            event="fetched all data from MARS",
+            filename=fi.filename(),
+            filepath=tfp.as_posix(),
+            nbytes=tfp.stat().st_size
+        )
+
+        return fi, tfp
+
+    def mapTemp(self, *, p: pathlib.Path) -> xr.Dataset:
+
+        log.debug(
+            event="mapping raw file to xarray dataset",
+            filepath=p.as_posix()
+        )
+
+        # Load the wholesale file as a list of datasets
+        # * cfgrib loads multiple hypercubes for a single multi-parameter grib file
+        # * Can also set backend_kwargs={"indexpath": ""}, to avoid the index file
+        try:
+            datasets: list[xr.Dataset] = cfgrib.open_datasets(
+                path=p.as_posix(),
+                chunks={
+                    "time": 1,
+                    "step": -1,
+                    "variable": -1,
+                    "longitude": "auto",
+                    "latitude": "auto"
+                },
+            )
+        except Exception as e:
+            log.warn(
+                event="error converting raw file to dataset",
+                filepath=p.as_posix(),
+                error=e
+            )
+            return xr.Dataset()
+
+        for i, ds in enumerate(datasets):
+
+            # Rename the parameters to the OCF names
+            # * Only do so if they exist in the dataset
+            for oldParamName, newParamName in PARAMETER_RENAME_MAP.items():
+                if oldParamName in ds:
+                    ds = ds.rename({oldParamName: newParamName})
+
+            # Snow depth is in `m` from ECMWF, but OCF expects `kg m-2`.
```

Comment:
do we need to do this?

Reply:
Well their variable is measured in m, and we do this to go from m's with the CEDA data. I thought there was a goal here to conform all our data?

Reply:
hmm, yea, hmm, not sure. Have you been checking our units or has this one just popped up?
@jacobbieker what do you think?

Reply:
It probably doesn't matter when it comes down to it on the ML side, as everything gets normalised on the way in anyway if I understand correctly! It's just for the sake of comparison between NWP sources - if we ever do that.

Reply:
yea, I'd be tempted to leave it how it is, to try to avoid any confusion later on. what do you think?

Reply:
As in, not do the multiplication?

Reply:
yea leave it in ECMWF units